frame/babe/src/lib.rs

```diff
@@ -619,14 +649,19 @@ impl<T: Config> Pallet<T> {
 	fn deposit_randomness(randomness: &schnorrkel::Randomness) {
 		let segment_idx = SegmentIndex::<T>::get();
 		let mut segment = UnderConstruction::<T>::get(&segment_idx);
-		if segment.len() < UNDER_CONSTRUCTION_SEGMENT_LENGTH {
+		if segment.len() < T::MaxSegmentLength::get() as usize {
```

Comment:
This if-else can just be rewritten altogether:

```rust
if segment.try_push(*randomness).is_ok() {
    UnderConstruction::<T>::insert(&segment_idx, &segment);
} else {
    // ...
}
```

Reply:
Good point, rewritten